src/lib/core/cord_on_demand.cc

```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2022, Tarantool AUTHORS, please see AUTHORS file.
+ */
+
+#include "cord_on_demand.h"
+
+#include <stdlib.h>
+
+#include "fiber.h"
+#include "trivia/util.h"
+
+/**
+ * RAII wrapper around a cord object.
+ *
+ * A thread-local cord object is created on the first call to the get() method
+ * and destroyed when the thread exits.
+ */
+class CordOnDemand final {
+public:
+	static cord *get()
+	{
+		static thread_local CordOnDemand singleton;
```

Comment:
Nit: `static` is redundant, `thread_local` already implies it, see https://timsong-cpp.github.io/cppwp/n3337/dcl.stc#4.

Reply:
Removed.